mmdet/datasets/utils.py

```diff
@@ -98,3 +104,44 @@ def get_loading_pipeline(pipeline):
         'The data pipeline in your config file must include ' \
         'loading image and annotations related pipeline.'
     return loading_pipeline_cfg
+
+
+@HOOKS.register_module()
+class NumClassCheckHook(Hook):
+
+    def _check_head(self, model, dataset):
+        """Check whether the `num_classes` in head matches the length of
+        `CLASSSES` in `dataset`.
+
+        Args:
+            model (obj:`MMDataParallel` | obj:`nn.Module`): One stage
+                or Two stage Detector
+            dataset (obj:`CustomDataset`): Detection dataset.
+        """
+
+        for name, module in model.named_modules():
+            if hasattr(module, 'num_classes') and not isinstance(
+                    module, (RPNHead, VGG, FusedSemanticHead, GARPNHead)):
+                assert module.num_classes == len(dataset.CLASSES), \
```

Comment:
In some cases `dataset.CLASSES` is equal to `None`.

Reply:
for example ?

Reply:
https://github.com/open-mmlab/mmdetection/blob/247785ceb31cabb0e3fd4c6ce17108d04f478ab8/mmdet/datasets/custom.py#L53